npm/pkg/dataplane/ipsets/dirtycache_linux.go

```diff
@@ -0,0 +1,119 @@
+package ipsets
+
+import "fmt"
+
+type memberAwareDirtyCache struct {
+	// map of prefixed set names to original members
+	toAddOrUpdateCache map[string]map[string]struct{}
+	// map of prefixed set names to original members
+	toDeleteCache map[string]map[string]struct{}
+}
+
+func newDirtyCache() dirtyCacheMaintainer {
+	return &memberAwareDirtyCache{
+		toAddOrUpdateCache: make(map[string]map[string]struct{}),
+		toDeleteCache:      make(map[string]map[string]struct{}),
+	}
+}
+
+func (dc *memberAwareDirtyCache) reset() {
+	dc.toAddOrUpdateCache = make(map[string]map[string]struct{})
+	dc.toDeleteCache = make(map[string]map[string]struct{})
+}
+
+func (dc *memberAwareDirtyCache) create(newSet *IPSet) {
+	setName := newSet.Name
+	if _, ok := dc.toAddOrUpdateCache[setName]; ok {
+		return
+	}
+	info, ok := dc.toDeleteCache[setName]
+	if !ok {
+		info = make(map[string]struct{})
+	}
+	dc.toAddOrUpdateCache[setName] = info
+	delete(dc.toDeleteCache, setName)
+}
+
+func (dc *memberAwareDirtyCache) update(originalSet *IPSet) {
+	putIntoAndRemoveFromOther(originalSet, dc.toAddOrUpdateCache, dc.toDeleteCache)
+}
+
+func (dc *memberAwareDirtyCache) delete(originalSet *IPSet) {
+	putIntoAndRemoveFromOther(originalSet, dc.toDeleteCache, dc.toAddOrUpdateCache)
+}
+
+func putIntoAndRemoveFromOther(originalSet *IPSet, intoCache, fromCache map[string]map[string]struct{}) {
+	setName := originalSet.Name
+	if _, ok := intoCache[setName]; ok {
+		return
+	}
+	members, ok := fromCache[setName]
+	if !ok {
```

Comment:
also if set is present in fromCache, then we will need to update members with current cache members